src/VisualStudio/Core/Def/ExternalAccess/VSTypeScript/Api/VSTypeScriptContainedLanguageWrapper.cs

```diff
@@ -40,6 +41,33 @@ internal struct VSTypeScriptContainedLanguageWrapper
                 vbHelperFormattingRule: null);
         }
 
+        public VSTypeScriptContainedLanguageWrapper(
+            IVsTextBufferCoordinator bufferCoordinator,
+            IComponentModel componentModel,
+            VSTypeScriptVisualStudioProjectWrapper project,
+            IVsHierarchy hierarchy,
+            uint itemid,
+            Guid languageServiceGuid)
+        {
+            if (project.IsDefault)
+            {
+                throw new ArgumentException($"A non-default {nameof(VSTypeScriptVisualStudioProjectWrapper)} is required.");
+            }
+
+            var workspace = componentModel.GetService<VisualStudioWorkspace>();
+            var filePath = ContainedLanguage.GetFilePathFromHierarchyAndItemId(hierarchy, itemid);
+
+            _underlyingObject = new ContainedLanguage(
+                bufferCoordinator,
+                componentModel,
+                workspace,
+                project.Project!.Id, // ! is safe due to IsDefault check.
```

Comment:
You may be able to annotate your VSTypeScriptVisualStudioProjectWrapper with MemberNotNull annotations so this can be checked by the compiler. Basically you put that on the IsDefault member and list the things that aren't null as a result.